doc/semantics.rst

```diff
@@ -46,16 +46,42 @@ Opposed to that, components maintained by a third party are declared through the
 Identities and Accessors
 ------------------------
 
-Sources and Resources are unambiguously identified by two-tuples of name and version, in the
-context of their declaring component version.
+Component versions are unambiguously identified by two-tuples of `name` and `version` in the
+context of a `Component Descriptor Context Repository`. Their identity never changes across
+any replication between context repositories.
+
+Components declare dependencies towards `Sources`, `Resources` and other `Components`.
+
+Any of those dependency declarations have an umambiguous identity in the context of their
+declaring component version. Their identities always consist of their formal type
```

Comment:
we should a a description of the identity format.
So say that there is an attributes called `extraIdEntities` that is a `string: string` tuple.
Whereas the string must only consist of ascii characters.

Reply:
yes. however, I figured it might make sense to separate the documentation that deals with the semantics and the syntactical definition. for the latter, I thought we might resort to the json-schema. wdyt?

the rationale behind that being that it is always hard to keep source-code and documentation in sync. therefore, I preferred to keep the natural-language-documentation less concrete, in favour if json-schema + code

Reply:
this is fine to me